extensions/amp-analytics/0.1/events.js

```diff
@@ -386,6 +405,84 @@ export class CustomEventTracker extends EventTracker {
   }
 }
 
+export class AmpStoryEventTracker extends CustomEventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    // TODO(artezan): add support for storySpec.
+    const rootTarget = this.root.getRootElement();
+
+    // Fire buffered events if any.
+    const buffer = this.buffer_ && this.buffer_[eventType];
+    if (buffer) {
+      const bufferLength = buffer.length;
+
+      for (let i = 0; i < bufferLength; i++) {
+        const event = buffer[i];
+        this.fireListener_(event, rootTarget, config, listener);
+      }
+    }
+
+    let observables = this.observables_[eventType];
+    if (!observables) {
+      observables = new Observable();
+      this.observables_[eventType] = observables;
+    }
+
+    return this.observables_[eventType].add(event => {
+      this.fireListener_(event, rootTarget, config, listener);
+    });
+  }
+
+  /**
+   * Fires listener given the specified configuration.
+   * @param {!AnalyticsEvent} event
+   * @param {!Element} rootTarget
+   * @param {!JsonObject} config
+
+   * @param {function(!AnalyticsEvent)} listener
+   */
+  fireListener_(event, rootTarget, config, listener) {
+    const type = event['type'];
+    const on = config['on'];
+
+    if (type !== on) {
```

Comment:
Looks like we've already filtered based on eventType, could you please explain why we need the check here? Thanks.